core/client/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -642,24 +642,34 @@ public boolean rename(Path src, Path dst) throws IOException {
 
     AlluxioURI srcPath = new AlluxioURI(HadoopUtils.getPathWithoutScheme(src));
     AlluxioURI dstPath = new AlluxioURI(HadoopUtils.getPathWithoutScheme(dst));
-    ensureExists(srcPath);
-    URIStatus dstStatus;
-    try {
-      dstStatus = mFileSystem.getStatus(dstPath);
-    } catch (IOException | AlluxioException e) {
-      dstStatus = null;
-    }
-    // If the destination is an existing folder, try to move the src into the folder
-    if (dstStatus != null && dstStatus.isFolder()) {
-      dstPath = dstPath.join(srcPath.getName());
-    }
     try {
       mFileSystem.rename(srcPath, dstPath);
-      return true;
-    } catch (IOException | AlluxioException e) {
+    } catch (AlluxioException e) {
```

Comment:
Based on the Alluxio exception thrown, we should be able to rule out some possibilities, for example if `FileDoesNotExist` is thrown, we should be able to fail immediately.